pennylane/tape/qscript.py

```diff
@@ -695,7 +695,14 @@ def set_parameters(self, params, trainable_only=True):
 
         for idx, p in iterator:
             op = self._par_info[idx]["op"]
-            op.data[self._par_info[idx]["p_idx"]] = p
+            if isinstance(
+                op, Tensor
+            ):  # the setter for tensor does not support the definition syntax
+                op.data[self._par_info[idx]["p_idx"]] = p
```

Comment:
Should we instead try and add a data setter for `Tensor`?

Reply:
moving to another PR, I'll try it out there.